src/Api/Controllers/TopicsController.Attachments.cs

```diff
@@ -127,22 +123,22 @@ public IActionResult PostAttachment([FromRoute]int topicId, [FromBody]Attachment
         [ProducesResponseType(typeof(void), 400)]
         [ProducesResponseType(typeof(void), 403)]
         [ProducesResponseType(typeof(EntityResult), 500)]
-        public IActionResult PutAttachment([FromRoute]int topicId, [FromRoute] int attachmentId, [FromForm]IFormFile file)
+        public async Task<IActionResult> PutAttachmentAsync([FromRoute]int topicId, [FromRoute] int attachmentId, [FromForm]IFormFile file)
         {
-            if (!_topicPermissions.IsAssociatedTo(User.Identity.GetUserIdentity(), topicId))
-                return Forbidden();
+            if (!(await _topicPermissions.IsAssociatedToAsync(User.Identity.GetUserIdentity(), topicId)))
+                return Forbid();
 
             if (file == null)
                 ModelState.AddModelError("file", "File is null");
 
             if (!ModelState.IsValid)
                 return BadRequest(ModelState);
 
-            var result = _attachmentsManager.PutAttachment(attachmentId, User.Identity.GetUserIdentity(), file);
+            var result = await _attachmentsManager.PutAttachmentAsync(attachmentId, User.Identity.GetUserIdentity(), file);
 
-            if (result.Success)
-                return Ok(result);
-            return NotFoundError(result);
+            return result.Success
+                ? Ok(result) as IActionResult
```

Comment:
Because `Ok()` and `NotFound()` return different types and the compiler is not smart enough to determine their common base type `IActionResult`. But due to your previous suggestion, this is now no longer needed anyways.